docs/pages/machine-id/reference/github-actions.mdx

```diff
@@ -0,0 +1,104 @@
+---
+title: GitHub Actions
+description: Reference for GitHub Actions joining
+---
+
+## Guides
+
+You can read step-by-step guides on using Machine ID and GitHub Actions:
+
+- [GitHub Actions and SSH](../guides/github-actions.mdx)
+- [GitHub Actions and Kubernetes](../guides/github-actions-kubernetes.mdx)
+
+## GitHub join token
+
+The `token` resource sets out rules for what is allowed to join a Teleport
+cluster. Joining clients must specify which `token` they want to use, and then
+information included in their join request is compared to the rules contained
+within the token by the Auth Server to determine whether or not they should be
+admitted.
+
+The following snippet shows all available options in the `token` resource when
+used for GitHub joining with Machine ID:
+
+```yaml
```

Comment:
Unsure if providing the field documentation here as comments makes sense, but I couldn't find a way of structuring it outside of this which didn't make it a total scroll-fest or turn it into a table with some very wordy entries. I'm easy on changing it to a table or a series of heading if thats preferred though, it won't take long to do.